packages/framer-motion/src/animation/optimized-appear/handoff.ts

```diff
@@ -0,0 +1,37 @@
+import { sync } from "../../frameloop"
+import { transformProps } from "../../render/html/utils/transform"
+import { appearStoreId } from "./store-id"
+
+export function handoffOptimizedAppearAnimation(
+    id: string,
+    name: string
+): number {
+    const { MotionAppearAnimations } = window
+
+    const animationId = appearStoreId(
+        id,
+        transformProps.has(name) ? "transform" : name
+    )
+    const animation =
+        MotionAppearAnimations && MotionAppearAnimations.get(animationId)
+
+    if (animation) {
+        /**
+         * We allow the animation to persist until the next frame:
+         *   1. So it continues to play until Framer Motion is ready to render
+         *      (avoiding a potential flash of the element's original state)
+         *   2. As all independent transforms share a single transform animation, stopping
+         *      it synchronously would prevent subsequent transforms from handing off.
```

Comment:
I didn't quite understand this comment, could you give an example?

Reply:
in psuedo code

```
animate("scale", 2)
 -> handoff from "transform" animation
 -> cancel "transform" animation
animate("x", 2)
 -> no "transform" animation present, start new animation
```

VS

```
animate("scale", 2)
 -> handoff from "transform" animation
 -> schedule cancel "transform" animation
animate("x", 2)
 -> handoff from "transform" animation
 -> schedule cancel "transform" animation
cancel "transform" animation
```